py/selenium/webdriver/remote/remote_connection.py

```diff
@@ -280,6 +301,13 @@ def __init__(self, remote_server_addr: str, keep_alive: bool = False, ignore_pro
             self._conn = self._get_connection_manager()
         self._commands = remote_commands
 
+    extra_commands = {}
+
+    @classmethod
+    def add_command(cls, name, method, url):
```

Comment:
Don't we need to also allow defining custom methods to access these commands? That's what we do at least in Ruby

https://github.com/SeleniumHQ/selenium/blob/2dcac9f34589711d6183ac7fa12f2c31effeba97/rb/spec/unit/selenium/webdriver/remote/bridge_spec.rb#L42-L51

Reply:
Oh, i didn't notice here.

Maybe this is for https://github.com/appium/python-client/blob/7ac6bb833022b7dd6c753fd806904ab9f3e9fb79/appium/webdriver/extensions/log_event.py#L69-L70 which is a bit different from the ruby's one. Let me use this in https://github.com/appium/python-client/pull/1045 later


The Ruby's one you thought is maybe https://github.com/appium/python-client/blob/7ac6bb833022b7dd6c753fd806904ab9f3e9fb79/appium/webdriver/webdriver.py#L65-L172

Reply:
Can you explain what needs to be done here for python? I will implement it.

Reply:
https://github.com/appium/python-client/blob/7ac6bb833022b7dd6c753fd806904ab9f3e9fb79/appium/webdriver/webdriver.py#L65-L172 this docstring's one. A user can add their arbitrary command as a new method. Personally, in Python's method, selneium's modification is not necessary since the style is giving each command class. A bit different from other bindings.


Hm, `add_command` could be a public instance method for this usage
e.g.
https://github.com/appium/python-client/pull/1045/files#diff-9eb2a344105c9330ef01acd17b3dc5ad00c9d2abc6422dbf3f63123ba48fc602

Reply:
@KazuCocoa I have changed the `add_command` method to an instance method and made other necessary changes to tests.